### PR TITLE
[travis] Support newer gcc.

### PR DESCRIPTION
**Pull request by rdzhabarov**
Thanks @artemrakhov for helping and reproducing the original issue on Linux.

### CODE REVIEW

**Comment by bertmaher** [comment]
Awesome!

File: .travis.yml [hunk 1]
```diff
@@ -44,12 +44,16 @@ jobs:
       env:
         - TEST_NAME=COVERAGE
       before_install:
+        - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
+        - sudo apt-get update -qq
         - utils/install_protobuf.sh
         - sudo apt-get install -y lcov
         - sudo pip install awscli --upgrade
       install:
+        - sudo apt-get install -qq g++-6
```
Comment (artemrakhov) [inline .travis.yml]: Why is this line not in before_install, together with other apt-get stuff?

File: .travis.yml [hunk 2]
```diff
@@ -44,12 +44,16 @@ jobs:
       env:
         - TEST_NAME=COVERAGE
       before_install:
+        - sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
```
Comment (artemrakhov) [inline .travis.yml]: I think we already have `ubuntu-toolchain-r-test` in sources. Please look at line 9

File: .travis.yml [hunk 3]
```diff
@@ -44,12 +44,16 @@ jobs:
...
+        - sudo apt-get update -qq
```
Comment (artemrakhov) [inline .travis.yml]: Not sure if this is needed at all.

File: .travis.yml [hunk 4]
```diff
@@ -44,12 +44,16 @@ jobs:
...
         - utils/install_protobuf.sh
         - sudo apt-get install -y lcov
         - sudo pip install awscli --upgrade
       install:
+        - sudo apt-get install -qq g++-6
         - mkdir build && cd build
-        - cmake -G Ninja
+        - sudo update-alternatives --install /usr/bin/gcov gcov /usr/bin/gcov-6 90
```
Comment (artemrakhov) [inline .travis.yml]: I'd also move this to before_install

**Pull request merged**